tests/ui/pattern/issue-76342.rs

```diff
@@ -0,0 +1,25 @@
+// check-pass
+
```

Comment:
ah, two more things:

- can you rename this file to `slice-patterns-irrefutable.rs`?
- can you add the following function to this test
```rust
use std::array::TryFromSliceError;
fn array_try_from(x: &[usize]) -> Result<usize, TryFromSliceError> {
    let [a, b] = x.try_into()?;
    Ok(a + b)
}
```